server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -697,6 +698,19 @@ func (e *EventChannel) FinalizeInvocation(iid string) error {
 		invocation.LastChunkId = e.logWriter.GetLastChunkId(ctx)
 	}
 
+	if e.beValues.ProfileURI() != nil {
+		w, err := e.env.GetBlobstore().Writer(ctx, invocation.InvocationId+"/"+strconv.FormatUint(invocation.Attempt, 10)+"/"+e.beValues.ProfileName())
```

Comment:
Instead of storing in an attempt-specific subdir, wdyt about associating it only with the invocation ID? e.g. maybe a dir like

```
/{invocation-id}/artifacts/{artifact-name}
```

This has a couple of advantages -
* The BES stream should be identical for each Bazel invocation (*) so if we were to use an attempt-specific dir then we'd potentially be storing duplicate artifacts.
* Having `attempt_number` forces us to do an invocation DB lookup in order to retrieve the artifact which slightly hurts performance, vs. just having the invocation ID which is readily available

(*) This is not true of synthetic workflow invocations since the CI runner task can potentially get cancelled and re-invoked, but this doesn't matter for now since we don't upload timing profiles/test logs for the synthetic workflow invocation.

Reply:
Sure, fine with me!